MockPyth.sol

```diff
@@ -7,21 +7,50 @@ import "./PythStructs.sol";
 contract MockPyth is AbstractPyth {
     mapping(bytes32 => PythStructs.PriceFeed) priceFeeds;
 
+    uint singleUpdateFeeInWei;
+    uint singleUpdateFeeInPythToken;
+
+    constructor(uint _singleUpdateFeeInWei, uint _singleUpdateFeeInPythToken) {
+        singleUpdateFeeInWei = _singleUpdateFeeInWei;
+        singleUpdateFeeInPythToken = _singleUpdateFeeInPythToken;
+    }
+
     function queryPriceFeed(bytes32 id) public override view returns (PythStructs.PriceFeed memory priceFeed) {
         return priceFeeds[id];
     }
 
     // Takes an array of encoded price feeds and stores them.
     // You can create this data either by calling createPriceFeedData or
     // by using web3.js or ethers abi utilities.
-    function updatePriceFeeds(bytes[] memory updateData) public override {
+    function updatePriceFeeds(bytes[] memory updateData, uint feeAmount, bool feeUsingPythToken) public override payable {
+        require(feeAmount >= getMinUpdateFee(updateData.length, feeUsingPythToken), "Insufficient fee amount");
+
+        if (feeUsingPythToken) {
+            // It doesn't do the actual transfer here, to avoid extra dependency.
+            // SafeERC20.safeTransferFrom(IERC20(pythToken), msg.sender, address(this), feeAmount);
```

Comment:
oh man so if we really wanted to do this, then we would have to make every user of Pyth approve this contract to transfer ERC20 wrapped Pyth tokens?

Reply:
Yep it's true. In the case of using Pyth token, I think only protocols pay Pyth token not their users.